controllers/storagecluster/provider_server.go

```diff
@@ -189,6 +192,38 @@ func (o *ocsProviderServer) ensureDeploymentReplica(actual, desired *appsv1.Depl
 	return nil
 }
 
+// updateStorageProviderEndpointInStatus updates the status.storageProviderEndpoint which is required for consumer to establish a connection with provider
+func (r *StorageClusterReconciler) updateStorageProviderEndpointInStatus(instance *ocsv1.StorageCluster, nodePort int32) error {
+
+	nodes := &corev1.NodeList{}
+
+	err := r.Client.List(context.TODO(), nodes)
+	if err != nil {
+		r.Log.Error(err, "Failed to list nodes")
+		return err
+	}
+
+	nodeAddress := []string{}
+
+	for _, node := range nodes.Items {
+		if _, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/worker"]; ok {
+			for _, address := range node.Status.Addresses {
+				if address.Type == corev1.NodeInternalIP {
+					nodeAddress = append(nodeAddress, address.Address)
```

Comment:
Any reasons for using `nodeAddress` as `[]string`.   If only first entry is needed, then maybe we can use it as string and break the loop here.

Reply:
I am doing this on purpose. As I am not sure about the order of nodes as it is a list. I am getting all of them and sorting it so that I will always use the same one.

Reply:
I am taking a single address only from each node as I have an if condition and there will be only one `NodeInternalIP`, I will put a `break`

Reply:
made changes